src/rdpc/analysesProcessor.ts

```diff
@@ -54,9 +53,15 @@ export const analysisStream_withSpeicmens = async function* ({
       donorId,
     });
 
+    // in case of api returns less analyses than chunk size, we need to stream from the last analysis
+    // to make sure there is no data loss:
+    if (page && page.length < chunkSize) {
+      chunkSize = page.length;
+    }
```

Comment:
We don't need to change the value of chunkSize, thats should have the value from our config. Instead, in the next line where we update `currentPage` just add whichever is smaller.
`streamState.currentPage += page.length < chunkSize ? page.length : chunkSize;